src/engine/contactwriter.cpp

```diff
@@ -3071,6 +3071,10 @@ static bool removeContactDetails(QContact *contact, const QList<StringPair> &rem
     return true;
 }
 
+// Despite the similarity in name between this function and modifyContactDetail,
+// this function is used when saving modifications to an aggregate contact.
+// Those updates get downpromoted to the appropriate constituent contact in this
+// function.
 static bool modifyContactDetails(QContact *contact, const QList<QPair<StringPair, QContactDetail> > &updates)
```

Comment:
Well, actually, the it modifies the constituent contact details, based upon the updates which were originally performed on the aggregate.  So, I think the function name is ok.  But I managed to confuse myself as to its meaning, and break something (luckily we had the unit test to catch that!) because of my misunderstanding; hence the comment.